test/upgrade/check-ssl-from-any_version-ssl.td

```diff
@@ -0,0 +1,17 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+> SELECT * FROM data
+a
+---
+1
+
+# ensure we actually have the new option
+> SHOW CREATE SOURCE data
+"materialize.public.data" "CREATE SOURCE \"materialize\".\"public\".\"data\" FROM KAFKA BROKER 'kafka:9092' TOPIC 'testdrive-data-${testdrive.seed}' WITH (\"security_protocol\" = 'SSL', \"ssl_key_location\" = '/share/secrets/materialized.key', \"ssl_certificate_location\" = '/share/secrets/materialized.crt', \"ssl_ca_location\" = '/share/secrets/ca.crt', \"ssl_key_password\" = 'mzmzmz') FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY 'https://schema-registry:8081/' SEED VALUE SCHEMA '{\"type\":\"record\",\"name\":\"envelope\",\"fields\":[{\"name\":\"before\",\"type\":[{\"type\":\"record\",\"name\":\"row\",\"fields\":[{\"name\":\"a\",\"type\":\"long\"}]},\"null\"]},{\"name\":\"after\",\"type\":[\"row\",\"null\"]}]}' WITH (\"ssl_ca_location\" = '/share/secrets/ca.crt', \"ssl_key_location\" = '/share/secrets/materialized.key', \"ssl_certificate_location\" = '/share/secrets/materialized.crt') ENVELOPE DEBEZIUM"
```

Comment:
Hm, this doesn't need to be `current_source`?

Reply:
Oh, I think I just forgot how the upgrade test works (again). The any_version here is about the FROM version, not the TO version. And yes, this migration should apply when you're coming FROM any old version of Materialize. 

I think rather than changing this to `current_source`, which I suggested in my sleepiness, you want to make this match `create-ssl-in-0.16.0` and use 0.16.0, so that we get maximum coverage of upgrading from old versions.

Reply:
ah, its counter-intuitive, `any_version` means do a check with the current source-built materialize on EVERY create-in-X-version

Reply:
I guess any_version means that, though? Confusing. Whatever you have seems to be fine!

Reply:
Wait, hang on, I think it'll break nightlies. Gonna disable auto-merge and DM ya.